src/Symfony/Component/DependencyInjection/Compiler/InlineServiceDefinitionsPass.php

```diff
@@ -38,77 +33,39 @@ public function setRepeatedPass(RepeatedPass $repeatedPass)
     }
 
     /**
-     * Processes the ContainerBuilder for inline service definitions.
-     *
-     * @param ContainerBuilder $container
+     * {@inheritdoc}
      */
-    public function process(ContainerBuilder $container)
+    protected function processValue($value, $isRoot = false)
     {
-        $this->compiler = $container->getCompiler();
-        $this->formatter = $this->compiler->getLoggingFormatter();
-        $this->graph = $this->compiler->getServiceReferenceGraph();
-
-        $container->setDefinitions($this->inlineArguments($container, $container->getDefinitions(), true));
-    }
+        if ($value instanceof ArgumentInterface) {
+            $this->processValue($value->getValues());
 
-    /**
-     * Processes inline arguments.
-     *
-     * @param ContainerBuilder $container The ContainerBuilder
-     * @param array            $arguments An array of arguments
-     * @param bool             $isRoot    If we are processing the root definitions or not
-     *
-     * @return array
-     */
-    private function inlineArguments(ContainerBuilder $container, array $arguments, $isRoot = false)
-    {
-        foreach ($arguments as $k => $argument) {
-            if ($isRoot) {
-                $this->currentId = $k;
-            }
-            if (is_array($argument)) {
-                $arguments[$k] = $this->inlineArguments($container, $argument);
-            } elseif ($argument instanceof ArgumentInterface) {
-                $argument->setValues($this->inlineArguments($container, $argument->getValues()));
-            } elseif ($argument instanceof Reference) {
-                if (!$container->hasDefinition($id = (string) $argument)) {
-                    continue;
-                }
+            return $value;
+        }
+        if ($value instanceof Reference && $this->container->hasDefinition($id = (string) $value)) {
+            $compiler = $this->container->getCompiler();
+            $definition = $this->container->getDefinition($id);
 
-                if ($this->isInlineableDefinition($id, $definition = $container->getDefinition($id))) {
-                    $this->compiler->addLogMessage($this->formatter->formatInlineService($this, $id, $this->currentId));
+            if ($this->isInlineableDefinition($id, $definition, $compiler->getServiceReferenceGraph())) {
+                $compiler->addLogMessage($compiler->getLoggingFormatter()->formatInlineService($this, $id, $this->currentId));
 
-                    if ($definition->isShared()) {
-                        $arguments[$k] = $definition;
-                    } else {
-                        $arguments[$k] = clone $definition;
-                    }
+                if ($definition->isShared()) {
```

Comment:
no. Shared definitions are returned directly, while non-shared ones are passed to the parent processing